packages/mdc-list/index.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import MDCComponent from '@material/base/component';
+import {MDCListFoundation} from './foundation';
+import {strings} from './constants';
+
+/**
+ * @extends MDCComponent<!MDCistFoundation>
+ */
+export class MDCList extends MDCComponent {
+  /** @param {...?} args */
+  constructor(...args) {
+    super(...args);
+    /** @private {!Function} */
+    this.handleKeydown_;
+    /** @private {!Function} */
+    this.focusInEventListener_;
+    /** @private {!Function} */
+    this.focusOutEventListener_;
+  }
+
+  /**
+   * @param {!Element} root
+   * @return {!MDCList}
+   */
+  static attachTo(root) {
+    return new MDCList(root);
+  }
+
+  destroy() {
+    this.root_.removeEventListener('keydown', this.handleKeydown_);
+    this.root_.removeEventListener('focusin', this.focusInEventListener_);
+    this.root_.removeEventListener('focusout', this.focusOutEventListener_);
+  }
+
+  initialSyncWithDOM() {
+    this.handleKeydown_ = this.foundation_.handleKeydown.bind(this.foundation_);
+    this.focusInEventListener_ = this.foundation_.handleFocusIn.bind(this.foundation_);
+    this.focusOutEventListener_ = this.foundation_.handleFocusOut.bind(this.foundation_);
+    this.root_.addEventListener('keydown', this.handleKeydown_);
+    this.root_.addEventListener('focusin', this.focusInEventListener_);
+    this.root_.addEventListener('focusout', this.focusOutEventListener_);
+    this.layout();
+  }
+
+  layout() {
+    const direction = this.root_.getAttribute(strings.ARIA_ORIENTATION);
+    this.vertical = direction === strings.ARIA_ORIENTATION_VERTICAL;
+
+    // List items need to have at least tabindex=-1 to be focusable.
```

Comment:
you mean tabindex=0?

Reply:
No, tabindex=-1. `li` elements cannot be focused unless they have a `tabIndex` value. If it's 0, then they're in the tab order. If it's -1, they're not in the tab order, but they are focusable.